### PR TITLE
fix: fix regression #2846 #2842

### LINKED ISSUE #2842: regression in gumtree-spoon-ast-diff
See https://ci.inria.fr/sos/job/gumtree-spoon-ast-diff/

### LINKED ISSUE #2846: bug when parsing 'final'
Bad code:

```
Thread errThread = new Thread(() -> {
  // comment
  try (final InputStream input = new FileInputStream("file.txt")) {
    // some code
  } catch (IOException ex) {
    // some code
  }
}); 
```
Stack:

```
Error: Position of CtExtendedModifiers: [final] not found in
spoon.SpoonException: Position of CtExtendedModifiers: [final] not found in
    at spoon.support.compiler.jdt.PositionBuilder.setModifiersPosition(PositionBuilder.java:543)
    at spoon.support.compiler.jdt.PositionBuilder.buildPositionCtElement(PositionBuilder.java:254)
    at spoon.support.compiler.jdt.ContextBuilder.enter(ContextBuilder.java:100)
    at spoon.support.compiler.jdt.JDTTreeBuilder.visit(JDTTreeBuilder.java:1289)
    ....
```

If you add a space before the `final` or remove the comment ' // comment' , then the problem will disappear.

Good:

```
Thread errThread = new Thread(() -> {
  // comment
  try ( final InputStream input = new FileInputStream("file.txt")) {
    // some code
  } catch (IOException ex) {
    // some code
  }
});
```
Spoon from master.

### PR DESCRIPTION
**Pull request by monperrus**
fix #2846 
fix #2842

### CODE REVIEW

**Comment by monperrus** [comment]
ready.

File: src/main/java/spoon/support/compiler/jdt/PositionBuilder.java [hunk 1]
```diff
@@ -539,9 +538,6 @@ private void setModifiersPosition(CtModifiable e, int start, int end) {
 			}
 			start = o2;
 		}
-		if (explicitModifiersByName.size() > 0) {
-			throw new SpoonException("Position of CtExtendedModifiers: [" + String.join(", ", explicitModifiersByName.keySet()) + "] not found in " + String.valueOf(contents, start, end - start));
-		}
```
Comment (nharrand) [inline src/main/java/spoon/support/compiler/jdt/PositionBuilder.java]: Is there a legitimate reason why `final` modifier would remain in `explicitModifiersByName` here?

**Comment by monperrus** [comment]
because the position information is not perfect in all cases.

**Comment by nharrand** [comment]
OK. Works for me!

**Comment by nharrand** [comment]
@tdurieux would you be more comfortable with #2854 ?

**Comment by monperrus** [comment]
#2854 is good, thanks.

Yet, there is a philosophical argument here:
- are we able to have a working solution for 100% of cases? can we predict all cases?
- if one case is missing, should we break the execution by throwing an exception?

**Pull request merged**